python/cudf/cudf/core/udf/groupby_typing.py

```diff
@@ -0,0 +1,226 @@
+# Copyright (c) 2020-2023, NVIDIA CORPORATION.
+from typing import Any, Dict
+
+import numba
+from numba import cuda, types
+from numba.core.extending import (
+    make_attribute_wrapper,
+    models,
+    register_model,
+    type_callable,
+    typeof_impl,
+)
+from numba.core.typing import signature as nb_signature
+from numba.core.typing.templates import AbstractTemplate, AttributeTemplate
+from numba.cuda.cudadecl import registry as cuda_registry
+from numba.np import numpy_support
+
+# Disable occupancy warnings to avoid polluting output when there are few
+# groups.
+numba.config.CUDA_LOW_OCCUPANCY_WARNINGS = 0
```

Comment:
Updated such that a context manager object is used before the final kernel launch here.

Reply:
@wence- I noticed tests started failing throughout the codebase after introducing this, namely coming from numba warnings popping up everywhere. It turned out to be because we actually disable these warnings globally already [here](https://github.com/rapidsai/cudf/blob/branch-23.02/python/cudf/cudf/__init__.py#L99) during cuDF's import.